### ISSUE TITLE
Builds for Windows arm64

### ISSUE THREAD

**Issue by kdpuvvadi** [opentofu/opentofu#798]
### OpenTofu Version

```shell
OpenTofu v1.6.0-alpha3
on windows_amd64
```

### Use Cases

tofu for Windows systems running on arm64 socs

### Attempted Solutions

-

### Proposal

Add config to  `goreleaser` config or remove ignore for windows arm 64 build

```yaml
ignore:
  - goos: windows
    goarch: arm64
---


### References

_No response_

**Comment by janosdebugs** [comment]
Hey @kdpuvvadi thank you for this issue and apologies for the delayed reply. We discussed this issue in the core team and were wondering what the actual use case for this was? Windows on ARM seems to be very much in the early stages and adding a new architecture is a long-term commitment. (See also the discussion on #1130.) Do you have any project that currently doesn't work because OpenTofu on Windows/ARM does not exist?

**Comment by kdpuvvadi** [comment]
> Do you have any project that currently doesn't work because OpenTofu on Windows/ARM does not exist?
 
Our devs are pulling and building on their [windows dev kits](https://learn.microsoft.com/en-us/windows/arm/dev-kit/) and it would make the process bit smoothers with the binaries readily avaiable. 

and for runners arm64 binaries are anyway avaiable.

**Comment by janosdebugs** [comment]
@kdpuvvadi are you currently building OpenTofu for ARM yourself? Can you share what you are doing with OpenTofu on Win/ARM?

**Comment by kdpuvvadi** [comment]
> @kdpuvvadi are you currently building OpenTofu for ARM yourself?

Yes, 

Using it for mostly to check & validate the config on devs machine while writing new configs or modifying exisiting ones before opening prs.

**Comment by janosdebugs** [comment]
Hey @kdpuvvadi (and the wider community). Thank you for the input. As far as we can tell Windows on ARM has a very low footprint and supporting a new platform, especially one that we don't have dev hardware for in the core team, is quite a bit of effort. We'd like to ask everyone who would directly benefit from Win/ARM support to add a 👍 reaction to the issue so we can decide on the implementation priorities based on community need.

**Comment by virgofx** [comment]
Developers are now running Windows on ARM devices like Surface Laptops which have completely switched from x64 to ARM, and they need native binaries. The footprint is only going to grow (similar to the transition on MacOS) Supporting this would be huge for the community. Would it be possible to add experimental ARM support and help these devs get the performance they need?

**Comment by apparentlymart** [comment]
In addition to the previous points, I'd also note that a build of the `tofu` executable for `windows_arm64` would not be very useful unless all of the providers you intend to use are also publishing packages for that architecture.

OpenTofu's current plugin model uses native executables and so each platform needs its own plugin package. Therefore typically the provider developers need to "go first" and publish packages for a new target before we could publish a CLI executable that would depend on it, because otherwise we'd be releasing something that would generally just fail as soon as you run `tofu init`.

Since we're building our own packages for some providers that are otherwise available only from sources that OpenTofu isn't licensed to access, we are at least a few steps ahead with those and [already publishing `windows_arm64` packages for them](https://github.com/opentofu/scripts/blob/d37f7b9a0ae61e0eed5e1b3bd365bba9603ac91b/.github/workflows/release.yml#L115-L121), and so a `windows_arm64` build of OpenTofu would at least be able to use _those_. But for packages folks are directly publishing to the OpenTofu registry themselves, the original developer would need to add this as a new target in their own build and release process.

For those who are already using OpenTofu on `windows_arm64` using your own local builds: have you found that the providers you want to use generally have suitable packages available already? If there is already some momentum around provider developers publishing packages for this platform then that would be a good signal in favor of offering official releases of OpenTofu CLI too.

**Comment by cam72cam** [comment]
It appears that terraform is adding support for windows_arm64 in their latest alphas.  Potentially this may mark a shift in provider support for that platform.

**Comment by cam72cam** [comment]
However, none of the official hashicorp providers at this point seem to have a windows_arm64 release. This was tested using the latest terraform alpha and `terraform providers lock -platform=windows_arm64`

**Comment by apparentlymart** [comment]
The maintainers discussed this again today:

- Currently the maintainers of many providers in the ecosystem are motivated primarily by what Terraform supports rather than what OpenTofu supports, and so we had previously been reluctant to add `windows_arm64` support to OpenTofu because it would likely cause pressure on and tensions with provider developers about whether to add a new build that Terraform would not be able to use.
- However, it now appears that the next Terraform minor release will offer a `windows_arm64` build. Once that's confirmed by being included in a more finalied release (as opposed to just an alpha), we expect to add `windows_arm64` builds in whatever is the next OpenTofu minor release series, with the expectation that provider developers will then gradually add `windows_arm64` builds of their plugins in subsequent releases too.
- We're going to mark this as accepted, but also mark it as blocked for now until there's a non-alpha Terraform release with support for this architecture.

    It seems likely that Terraform v1.15 will include this, but we also have an understanding that this won't become subject to Terraform's compatibility promises until it's in a final release, and so there is some possibility that their plans will change in the meantime.
- No-one on the core maintainer team currently has access to `windows_arm64` systems for testing, but [GitHub Actions offers runners on that platform](https://github.blog/changelog/2024-09-03-github-actions-arm64-linux-and-windows-runners-are-now-generally-available/) so we should at least be able to run end-to-end tests and potentially also unit tests there.

    Perhaps we'll choose to at least partially resolve https://github.com/opentofu/opentofu/issues/3278 first so that e.g. we'd run `windows_arm64` _instead of_ `windows_386` tests, rather than running both. (But we can consider that more once this is unblocked.)

**Comment by apparentlymart** [comment]
Hi all,

The maintainers discussed this again today, because the Terraform team published their v1.15.0-beta1 which announces support for the `windows_arm64` target in Terraform, and so we could now add builds for that platform without causing friction for provider developers that are only testing with Terraform.

However, it seems that GitHub Actions support for hosted `windows_arm64` runners is still at a preview phase rather than generally available, and none of our maintainers have direct access to suitable hardware for local testing, and so for now we will not be able to make this an "officially supported" platform in the sense that we'll be routinely testing against it.

It will begin in the same "best-effort" category as the various other platforms without CI support, such as `solaris_amd64`, which means that:

- There is some risk of regressions in behavior of `windows_arm64` builds that we won't be able to catch in pre-release testing.
- If someone reports a bug that is specific to that platform then we expect to accept it unless it would make maintenance harder on other supported platforms, but we'll need to rely on members of the community to implement and test the fix on their own `windows_arm64` systems.
- A specific implication of the above is that fixes for issues specific to this platform may be slower to review and merge due to the additional coordination with community testers, and that's true even for security-related problems that are specific to this platform.

We'll revisit that support policy once GitHub Actions has generally-available support for runners on this platform, and once usage statistics suggest that it's being used at a similar scale to the other platforms that are already officially supported. But we hope that providing precompiled builds for this platform will be useful to those watching this issue despite the initial "best effort" status.

We are getting pretty close to the feature freeze for the v1.12.0 release so we might not be able to get this change in before that cutoff, but we'll try to get `windows_amd64` releases added soon. Thanks!

**Comment by fiodortretyakov** [comment]
Following up on the maintainers' last update here: with Terraform 1.15 now shipping the `windows_arm64` target (which removed the provider-ecosystem friction noted above), this looks ready to move forward — but it doesn't appear to have landed yet. `.goreleaser.yaml` on `main` still has:

```yaml
      - goos: windows
        goarch: arm64
```

in the `ignore:` list, and v1.12.1 ships no `tofu_*_windows_arm64.zip`.

I'd be happy to open the small PR to enable best-effort `windows_arm64` archives (dropping just that one `ignore` entry; leaving `windows/arm` 32-bit ignored), matching the best-effort, no-CI support tier described above, plus a CHANGELOG note. I can test the resulting build locally on a Snapdragon X / Windows 11 ARM64 machine.

Since the contribution guide asks for an `accepted` **and** `help-wanted` issue before opening a code PR, could a maintainer add `help-wanted` / assign this to me? Happy to proceed as soon as you give the go-ahead.

**Comment by cam72cam** [comment]
@apparentlymart I believe that https://github.blog/changelog/2025-08-07-arm64-hosted-runners-for-public-repositories-are-now-generally-available/ makes them available for public use, without restriction. It took a bit of digging to find the blog post, they don't mention the "status" in their main docs page.

**Comment by apparentlymart** [comment]
Yes, it seems like Windows on arm64 is now also listed in [the table of supported platforms for public repositories](https://docs.github.com/en/actions/reference/runners/github-hosted-runners#standard-github-hosted-runners-for-public-repositories), and so we could now revisit that earlier decision and consider offering full support for this platform.

**Comment by fiodortretyakov** [comment]
Created [https://github.com/opentofu/opentofu/pull/4234](https://github.com/opentofu/opentofu/pull/4234)
Please review

**Comment by apparentlymart** [comment]
Hi everyone!

The maintainers discussed this again today due to the news of GitHub Actions now having generally-available support for us to run tests on this platform.

Our updated decision is that we intend to add `windows_arm64` as a first-class-supported platform in the forthcoming v1.13 release series. This means that in addition to adding pre-built archives for this platform in our release process we will also:
- List `windows_arm64` under [Supported Platforms](https://opentofu.org/docs/language/v1-compatibility-promises/#supported-platforms) in our compatibility promises document, alongside `windows_amd64` and using similar wording as the existing item about Linux support.
- Introduce `windows_arm64` to the matrix of platforms where we run unit tests for every pull request.
- Prioritize fixing bugs specific to this platform to the same extent as we would bugs specific to any other of the officially-supported platforms.

All of those changes should be completed before we close this issue, but they don't necessarily all need to happen in the same pull request.
